filament/backend/src/Program.cpp

```diff
@@ -22,7 +22,11 @@ namespace filament {
 namespace backend {
 
 // We want these in the .cpp file so they're not inlined (not worth it)
-Program::Program() noexcept = default;
+#ifdef _MSC_VER
+    Program::Program() noexcept {}
+#else
+    Program::Program() noexcept = default;
```

Comment:
I'm baffled why MSVC doesn't allow this- it's perfectly valid C++11. Does it complain if you inline it in the class definition?

Reply:
The message says: "exception specification does not match the implicitly declared specification."
The implicitly declared specification does not have noexcept I believe.
If you want I can try inlining it in the class definition.

Reply:
cppreference says in https://en.cppreference.com/w/cpp/language/default_constructor:

> The implicitly-declared (or defaulted on its first declaration) default constructor has an exception specification as described in dynamic exception specification (until C++17)exception specification (since C++17)

In this case the constructor is *not* defaulted on its first declaration (inline in the class definition), and maybe that is why MSVC doesn't provide the nothrow exception specification.

In any case, I'm not sure what is the benefit in that case of `= default` vs providing an empty body.

Reply:
Full error is:

```
1>C:\greg\github\filament\filament\backend\src\Program.cpp(28,42): error C2610: 'filament::backend::Program::Program(void) noexcept': is not a special member function or comparison operator which can be defaulted
1>C:\greg\github\filament\filament\backend\src\Program.cpp(28,42): message : exception specification does not match the implicitly declared specification.
```

Reply:
> Does it complain if you inline it in the class definition?

No `Program() noexcept = default;` in the class definition works fine.

Reply:
I can update the PR if you prefer this solution.

Reply:
Inline in the definition or just `{}` everywhere would be fine.

Reply:
@romainguy done